### PR TITLE
Edit of docs/index page

### PR DESCRIPTION
**Pull request by sascala**

### CODE REVIEW

File: docs/docs/index.md [hunk 1]
```diff
@@ -35,33 +35,49 @@ Marathon packages are available from Mesosphere's [repositories](http://mesosphe
...
+As of version 0.9.0, Marathon adheres to [semantic versioning](http://semver.org).
+That means our documented REST API remains compatible across releases unless we change the MAJOR version
+(the first number in the version tuple). If you depend on undocumented features, please tell us about them by [raising a GitHub issue](https://github.com/mesosphere/marathon/issues/new). Portions of the API marked as EXPERIMENTAL are exempt from this rule. We do not introduce new features in PATCH version increments (the last number in the version tuple).
 
-We might change the command line interfaces of the Marathon server process in rare cases in a MINOR version upgrade.
-Please check the release notes for these.
+In rare cases, we may change the command line interfaces of the Marathon server process in a MINOR version upgrade. The release notes document these instances.
```
Comment (sascala) [inline docs/docs/index.md]: Is there another way to say "the command line interfaces of the Marathon server process"? Or can someone explain to me what this means?

File: docs/docs/index.md [hunk 2]
```diff
@@ -35,33 +35,49 @@ Marathon packages are available from Mesosphere's [repositories](http://mesosphe
...
-give us feedback (particularly on violations of the versioning policy).
+We provide release candidates for all new MAJOR/MINOR versions and invite our users to test them and
+give us feedback, particularly on violations of the versioning policy.
 
 ### Upgrading to a Newer Version
 
-[See here]({{ site.baseurl }}/docs/upgrade/index.html) for our guide on upgrading to a new version.
+See [the Marathon upgrade guide](https://mesosphere.github.io/marathon/docs/upgrade/index.html) to learn how to upgrade to a new version.
 
 ### Running in Production Mode
```
Comment (sascala) [inline docs/docs/index.md]: What is production mode exactly?

File: docs/docs/index.md [hunk 3]
```diff
@@ -35,33 +35,49 @@ Marathon packages are available from Mesosphere's [repositories](http://mesosphe
...
 
-To launch Marathon in *production mode*, you need to have both
-ZooKeeper and Mesos running. The following command launches
-Marathon in *production mode*. Point your web browser to
+Both ZooKeeper and Mesos need to be running in order to launch Marathon in *production mode*.
+
+Point your web browser to
```
Comment (sascala) [inline docs/docs/index.md]: What does this step have to do with production mode?

File: docs/docs/index.md (hunk 1 above)
Comment (gkleiman) [inline docs/docs/index.md]: This means that we might change the [command line flags](https://mesosphere.github.io/marathon/docs/command-line-flags.html) that are used to configure Marathon.

File: docs/docs/index.md (hunk 2 above)
Comment (gkleiman) [inline docs/docs/index.md]: I think that the right name should be High-Availability Mode.

cc/ @aquamatthias

File: docs/docs/index.md (hunk 3 above)
Comment (gkleiman) [inline docs/docs/index.md]: In HA mode, multiple Marathons will run at the same time, but only one of the instances will be the leader. The other instances will forward requests to the leader.

Marathon depends on ZooKeeper to perform the leader election.

**Comment by aquamatthias** [comment]
Close for now - please reopen when ready.

**Pull request closed without merging**